test/arbcall-test.jl

```diff
@@ -0,0 +1,130 @@
+                "fmpq_t x",
+                "mag_ptr res",
+                "arb_ptr res",
+                "acb_ptr res",
+                "const fmpz_t x",
+                "const fmpq_t x",
+                "mag_srcptr res",
+                "arb_srcptr res",
+                "acb_srcptr res",
+
+                # Internal types
+                "mpfr_rnd_t rnd",
+                "mp_limb_t lo",
+                "mp_bitcnt_t r",
+                "mp_ptr ycos",
+                "mp_srcptr x",
+                )
+        @test_throws KeyError arg = Arblib.Carg(str)
+    end
+
+    # Parsed incorrectly
+    for str in ("const char * inp",
+                )
+        @test_throws KeyError arg = Arblib.Carg(str)
+    end
+
+
+    # Parse errors
+    for str in (# Internal types
+                "mp_limb_t * error",
+                "mp_bitcnt_t * Qexp",
+                )
+        @test_throws ArgumentError arg = Arblib.Carg(str)
+    end
+end
+
+@testset "jlfname" begin
+    for (arbfname, name) in (# Supported types
+                             ("mag_set", :set),
+                             ("mag_set_d", :set),
+                             ("mag_set_ui", :set),
+                             ("arf_set", :set),
+                             ("arf_set_ui", :set),
+                             ("arf_set_si", :set),
+                             ("arf_set_d", :set),
+                             ("arb_set", :set),
+                             ("arb_set_arf", :set),
+                             ("arb_set_si", :set),
+                             ("arb_set_ui", :set),
+                             ("arb_set_d", :set),
+                             ("acb_set", :set),
+                             ("acb_set_ui", :set),
+                             ("acb_set_si", :set),
+                             ("acb_set_d", :set),
+                             ("acb_set_arb", :set),
+                             ("acb_set_si_si", :set),
+                             ("acb_set_d_d", :set),
+                             ("acb_set_arb_arb", :set),
+
+                             # Unsupported types
+                             ("arf_set_fmpz", :set_fmpz),
+                             ("arf_set_mpfr", :set_mpfr),
+                             ("acb_set_fmpq", :set_fmpq),
+                             ("arb_bin_uiui", :bin_uiui),
+
+                             # Deprecated types
+                             ("arf_set_mpz", :set_mpz),
+                             ("arf_set_fmpr", :set_fmpr),
+
+                             # Underscore methods
+                             ("_arb_vec_set", :_arb_vec_set),
+
+                             # Some special cases to be aware of and maybe change
+                             ("mag_set_d_lower", :set_d_lower),
+                             ("arb_ui_div", :ui_div),
+                             ("arb_rising_ui_rec", :rising_ui_rec),
+                             ("arb_zeta_ui_vec", :zeta_ui_vec),
```

Comment:
Right now, we only remove `ui` etc. if it appears at the end. However, we could also consider removing it at any position, i.e., `arb_zeta_ui_vec` -> `zeta_vec`.

Reply:
I think this is reasonable. The types should still all be different to there should not be any clashing name issues.

Reply:
let's do it then